src/backend/vm/runtime.rs

```diff
@@ -337,5 +404,113 @@ mod tests {
         } else {
             panic!("Wrong type");
         }
+        if let Data::Int(data) = vm.load_register(7).unwrap() {
```

Comment:
I think it would be better to abstract out "data equivalence tests". Please see @permui [abstract data equivalence tests](https://github.com/diatom-lang/diatom/pull/3/commits/06dca8f36218d67e991b3a2ff5331245ee50f6bd).